src/loci/formats/services/JZarrServiceImpl.java

```diff
@@ -76,20 +80,26 @@ public class JZarrServiceImpl extends AbstractService
    */
   public JZarrServiceImpl(String root) {
       checkClassDependency(com.bc.zarr.ZarrArray.class);
-      if (root != null && root.toLowerCase().contains("s3:")) {
-        LOGGER.warn("S3 access currently not supported");
+      if (root != null && (root.toLowerCase().contains("s3:") || root.toLowerCase().contains("s3."))) {
+        String[] pathSplit = root.toString().split(File.separator);
+        if (!S3FileSystemStore.ENDPPOINT_PROTOCOL.contains(pathSplit[0].toLowerCase())) {
+          s3fs = new S3FileSystemStore(Paths.get(root));
+        }
+        else {
+          LOGGER.warn("Zarr Reader is not using S3FileSystemStore as this is currently for use with S3 configured with a https endpoint");
+        }
       }
   }
 
   @Override
   public void open(String file) throws IOException, FormatException {
     currentId = file;
-    // TODO: Update s3 location identification
-    if (!file.toLowerCase().contains("s3:")) {
+    if (s3fs == null) {
       zarrArray = ZarrArray.open(file);
     }
     else {
-      LOGGER.warn("S3 access currently not supported");
+      s3fs.updateRoot(getZarrRoot(s3fs.getRoot()) + stripZarrRoot(file));
```

Comment:
The repetition of this call for all commands makes me wonder whether it should be part of the constructor?

Reply:
The update root is really setting the location of the particular zarr array to be opened in each case, so unfortunately it is needed in multiple locations.